File: 1667-find-kth-bit-in-nth-binary-string/find-kth-bit-in-nth-binary-string.py
class Solution:
    

    def findKthBit(self, n: int, k: int) -> str:
        if k==1:return '0'
        def rev(s):
            s=[l[i]for i in range(-1,-len(l)-1,-1)]
            for i in range(len(s)):
                if s[i]=='0':s[i]='1'
                else:s[i]='0'
            return s
        l=["0"]
        for i in range(n-1):
            s=list(l)
            s.append('1')
            s.extend(rev(s))
            l=s
        return s[k-1]
        
        
        
        
        
        

        

File: 1667-find-kth-bit-in-nth-binary-string/find-kth-bit-in-nth-binary-string.java
class Solution {
  
     public StringBuilder inv(StringBuilder s)
        { StringBuilder res = new StringBuilder(s.reverse());
        

            for(int i=0;i<res.length();i++)
            {
                if(res.charAt(i)=='0')
                {
                    res.setCharAt(i,'1');
                }
                else{
                     res.setCharAt(i,'0');
                }
            }
            return res;
        }
    public char findKthBit(int n, int k) {
        if(k==0) return '0';
        char c='a';
        StringBuilder s = new StringBuilder("0");
       for(int i=0;i<n;i++)
       {
        s.replace(0,s.length(),s.toString()+"1"+inv(s).toString());
         
       }
       
        return s.charAt(k-1);
    }
}